screens/Lectures/VideoLectureScreen.tsx

```diff
@@ -1,67 +1,63 @@
-import ReactSliderProgress from "./ReactSliderProgress";
-import PropTypes from "prop-types";
+import { useState, useEffect } from "react";
+import { View, TouchableOpacity, Alert, Text } from "react-native";
+import { Video, ResizeMode } from "expo-av";
 import { Icon } from "@rneui/themed";
 import { useNavigation } from "@react-navigation/native";
-import { completeComponent, handleLastComponent } from "../../services/utils";
-import { getVideoURL } from "../../services/storage-service";
+import { completeComponent, handleLastComponent } from "@/services/utils";
+import { getVideoURL } from "@/services/storage-service";
+import { Lecture } from "@/types/lecture";
+import { Course } from "@/types/course";
+
+export interface VideoLectureScreenProps {
+  lectureObject: Lecture;
+  courseObject: Course;
+  isLastSlide: boolean;
+  onContinue: () => void;
+  handleStudyStreak: () => void;
+}
 
 const VideoLectureScreen = ({
   lectureObject,
   courseObject,
   isLastSlide,
   onContinue,
   handleStudyStreak,
-}) => {
+}: VideoLectureScreenProps) => {
   const navigation = useNavigation();
-  const videoRef = useRef(null);
-  const [isPlaying, setIsPlaying] = useState(false);
-  const [positionMillis, setPositionMillis] = useState(0);
-  const [durationMillis, setDurationMillis] = useState(0);
-  const [isMuted, setIsMuted] = useState(false);
-  const [videoUrl, setVideoUrl] = useState(null);
-  const [showPlayPauseIcon, setShowPlayPauseIcon] = useState(false);
-  const [currentResolution, setCurrentResolution] = useState("360");
-  const [allResolutions] = useState(["1080", "720", "480", "360"]);
+  const [videoUrl, setVideoUrl] = useState<null | string>(null);
+  const currentResolution = "1080";
 
   // Fetch video URL based on current resolution
   useEffect(() => {
-    const fetchVideoUrl = async () => {
-      try {
-        const url = await getVideoURL(lectureObject.video, currentResolution);
-        if (!url) {
-          throw new Error("Video URL is null");
-        }
-        setVideoUrl(url);
-      } catch (error) {
-        console.error("Error fetching video URL:", error);
-        Alert.alert(
-          "Error",
-          "Failed to load the video. Please try again later.",
-          [{ text: "OK" }],
-        );
-        setVideoUrl(null);
-      }
+    const fetchVideoUrl = () => {
+      getVideoURL(lectureObject.content, currentResolution)
+        .then((url) => {
+          // Check if URL is a video
+          return fetch(url, { method: "HEAD" }).then((response) => {
+            const contentType: string | null =
+              response.headers.get("content-type");
+            if (!contentType?.startsWith("video/")) {
+              throw new Error("Invalid URL: " + url);
+            }
+            setVideoUrl(url);
+          });
+        })
+        .catch((error: unknown) => {
+          console.error(error);
+          setVideoUrl(null);
+        });
```

Comment:
Use `try/catch` with `await` instead, no `.then` and `.catch` chains. It was fine as it was before. You could just fix the logic a bit. And no `fetch` but use one of the two Axios clients from `axios.ts` if possible, or directly with `axios.get`.